### DISCUSSION TITLE
Power BI connector - GCC-High

### DISCUSSION THREAD

**Discussion by dzinn0203** [open-metadata/OpenMetadata#18492] [Q&A]
Can OM connect to a Power BI tenant located in a GCC-High cloud environment?

I've been trying to configure the connection but cannot get it to test successfully. I have setup a service principal in Azure, and I have another metadata management application using it to connect to Power BI and pull tenant info successfully, so I don't believe that is the problem.
Here are my connection settings:
![image](https://github.com/user-attachments/assets/68c5541e-be1e-4cfa-8f19-b405a3f08b4c)

This is the error I get every time:
![image](https://github.com/user-attachments/assets/ecd22082-c5c5-4706-a069-26b46722d1be)

I am on version 1.5.5.

Here is a snippet from the log:
  File "/home/airflow/.local/lib/python3.10/site-packages/metadata/ingestion/source/dashboard/dashboard_service.py", line 228, in __init__
    self.client = get_connection(self.service_connection)
  File "/home/airflow/.local/lib/python3.10/site-packages/metadata/ingestion/source/connections.py", line 54, in get_connection
    return get_connection_fn(connection)(connection)
  File "/home/airflow/.local/lib/python3.10/site-packages/metadata/ingestion/source/dashboard/powerbi/connection.py", line 40, in get_connection
    api_client=PowerBiApiClient(connection), file_client=file_client
  File "/home/airflow/.local/lib/python3.10/site-packages/metadata/ingestion/source/dashboard/powerbi/client.py", line 67, in __init__
    self.auth_token = self.get_auth_token()
  File "/home/airflow/.local/lib/python3.10/site-packages/metadata/ingestion/source/dashboard/powerbi/client.py", line 97, in get_auth_token
    raise InvalidSourceException(
metadata.ingestion.api.steps.InvalidSourceException: Failed to generate the PowerBi access token. Please check provided config

**Comment by Prajwal214** [comment]
Connecting OpenMetadata to a Power BI tenant in a GCC-High cloud environment requires specific configurations due to the unique endpoints and authentication mechanisms used in these environments. 

1. Correct Endpoints: Ensure that you are using the correct endpoints for GCC-High. The authentication URL should be https://login.microsoftonline.us/, and the Power BI API endpoint should be https://app.high.powerbigov.us/home/.

2. Service Principal Configuration: Double-check that the service principal is correctly configured with the necessary permissions. It should have the Dashboard.Read.All and Dataset.Read.All permissions granted in Azure AD.

3. Scope and Authority URI: Ensure that the scope is set to https://analysis.usgovcloudapi.net/powerbi/api/.default and the authorityUri is correctly configured to match the GCC-High environment.

4. Logs and Error Messages: Review the logs for any additional error messages that might provide more context. The error "Failed to generate the PowerBi access token" suggests an issue with the authentication configuration, so ensure all credentials (Client ID, Client Secret, Tenant ID) are correct.

**Reply by dzinn0203** [reply]
Great, thank you for the info. Since I posted this question, we've confirmed all of the above and have been able to get a test connection to complete successfully. But, when we try to execute a metadata ingestion, it fails. My colleague has posted an issue at [https://github.com/open-metadata/OpenMetadata/issues/18549](url). 

Essentially it appears that maybe the commercial Power BI URL is hard coded somewhere. The main error we are seeing is:
`{client.py:119} WARNING - Error fetching dashboards: 403 Client Error: Forbidden for url: https://api.powerbi.com/v1.0/myorg/admin/dashboards`

That URL is for the Power BI commercial cloud environment, not GCC-High, which we have configured in OM.
